hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/HoodieSparkCopyOnWriteTable.java

```diff
@@ -264,6 +269,24 @@ public Iterator<List<WriteStatus>> handleInsert(
     return Collections.singletonList(createHandle.close()).iterator();
   }
 
+  @Override
+  public boolean supportsFileGroupReader() {
+    return true;
+  }
+
+  @Override
+  public List<WriteStatus> runCompactionUsingFileGroupReader(String instantTime,
```

Comment:
Why we implement the compaction on COW table?

Reply:
The existing implementation of compaction uses `HoodieSparkCopyOnWriteTable#handleUpdate` which implements `HoodieCompactionHandler#handleUpdate`.  That's because `HoodieSparkMergeOnReadTable` uses `HoodieSparkCopyOnWriteTable` instance for compaction the following logic:
```
@Override
  public HoodieWriteMetadata<HoodieData<WriteStatus>> compact(
      HoodieEngineContext context, String compactionInstantTime) {
    RunCompactionActionExecutor<T> compactionExecutor = new RunCompactionActionExecutor<>(
        context, config, this, compactionInstantTime, new HoodieSparkMergeOnReadTableCompactor<>(),
        new HoodieSparkCopyOnWriteTable<>(config, context, getMetaClient()), WriteOperationType.COMPACT);
    return compactionExecutor.execute();
  }
```
So for parity I have to add the same here to not break any contract around `HoodieTable`.  I think we need to revisit this; the use of `HoodieSparkCopyOnWriteTable` instance in `HoodieSparkMergeOnReadTable` is counter-intuitive.